internal/kafka/internal/services/data_plane_kafka.go

```diff
@@ -52,67 +64,126 @@ func NewDataPlaneKafkaService(kafkaSrv KafkaService, clusterSrv ClusterService,
-			continue
-		}
-		var e *serviceError.ServiceError
-		switch s := getStatus(ks); s {
-		case statusReady:
-			// Store the routes (and create them) when Kafka is ready. By the time it is ready, the routes should definitely be there.
-			e = d.persistKafkaRoutes(kafka, ks, cluster)
-			if e == nil {
-				kafka.AdminApiServerURL = ks.AdminServerURI
-				e = d.setKafkaClusterReady(kafka)
-			}
-		case statusInstalling:
-			// Store the routes (and create them) if they are available at this stage to lessen the length of time taken to provision the Kafka.
-			// The routes list will either be empty or complete.
-			e = d.persistKafkaRoutes(kafka, ks, cluster)
-		case statusError:
-			// when getStatus returns statusError we know that the ready
-			// condition will be there so there's no need to check for it
-			readyCondition, _ := ks.GetReadyCondition()
-			e = d.setKafkaClusterFailed(kafka, readyCondition.Message)
-		case statusDeleted:
-			e = d.setKafkaClusterDeleting(kafka)
-		case statusRejected:
-			e = d.reassignKafkaCluster(kafka)
-		case statusRejectedClusterFull:
-			e = d.unassignKafkaFromDataplaneCluster(kafka)
-		case statusUnknown:
-			log.Infof("kafka cluster %s status is unknown", ks.KafkaClusterId)
-		default:
-			log.V(5).Infof("kafka cluster %s is still installing", ks.KafkaClusterId)
-		}
-		if e != nil {
-			log.Error(errors.Wrapf(e, "Error updating kafka %s status", ks.KafkaClusterId))
+
+	prewarmingStatusInfo := map[string]map[string]int{}
+	supportedInstanceTypes := cluster.GetSupportedInstanceTypes()
+
+	// prefill prewarming status info with empty count
+	for _, supportedInstanceType := range supportedInstanceTypes {
+		// sets all the statuses to 0 to allow for the count to drop
+		prewarmingStatusInfo[supportedInstanceType] = map[string]int{
+			statusReady.String():      0,
+			statusError.String():      0,
+			statusUnknown.String():    0,
+			statusInstalling.String(): 0,
 		}
+	}
 
-		e = d.setKafkaRequestVersionFields(kafka, ks)
-		if e != nil {
-			log.Error(errors.Wrapf(e, "Error updating kafka '%s' version fields", ks.KafkaClusterId))
+	for _, ks := range status {
+		deploymentType := getDeploymentType(ks)
+		switch deploymentType {
+		case realDeploymentType:
+			d.processRealKafkaDeployment(ks, cluster, log)
+		case reservedDeploymentType:
+			d.processReservedKafkaDeployment(ks, prewarmingStatusInfo, log, clusterID)
 		}
 	}
+
+	// updates the cluster prewarming status count
+	cluster.SetPrewarmingStatusInfo(prewarmingStatusInfo)
+	err = d.clusterService.Update(*cluster)
+	if err != nil {
+		log.Error(err)
+	}
+
 	return nil
 }
 
+// processRealKafkaDeployment process reserved kafka instances and updates their status in cluster
+func (d *dataPlaneKafkaService) processReservedKafkaDeployment(ks *dbapi.DataPlaneKafkaStatus, prewarmingStatusInfo map[string]map[string]int, log logger.UHCLogger, clusterID string) {
+	parsedID := strings.Split(ks.KafkaClusterId, "-")
```

Comment:
Isn't KafkaClusterId different than the name?

Reply:
What name?

Reply:
The name of the ManagedKafka.

The Kafka ID does not necessarly need to be the Kafka Name (same with the Cluster ID vs the Cluster name in the context of the Cluster type)

Reply:
Ah right. 
For reserved MK CR, the name and the id will be the same.
I always confuse the pattern and I think this method needs some reworking as the instance type will be the third element not the second.

The pattern is `reserved-kafka-<instanceType>-<number>`

Reply:
In that case. Shouldn't the length check be 4? :thinking:

Reply:
Yes, we could change it to 4. For the check added here, I was mainly interested in making sure that the third element (representing the instance-type) is there to avoid panic.

Reply:
changed to `4` as well to ensure that the format is respected and if not we've logs of why it wasn't respected.